server/src/main/java/io/druid/segment/indexing/DataSchema.java

```diff
@@ -39,33 +44,72 @@
   private static final Logger log = new Logger(DataSchema.class);
 
   private final String dataSource;
-  private final InputRowParser parser;
+  private final Map<String, Object> parser;
   private final AggregatorFactory[] aggregators;
   private final GranularitySpec granularitySpec;
 
+  private final ObjectMapper jsonMapper;
+
   @JsonCreator
   public DataSchema(
       @JsonProperty("dataSource") String dataSource,
-      @JsonProperty("parser") InputRowParser parser,
+      @JsonProperty("parser") Map<String, Object> parser,
       @JsonProperty("metricsSpec") AggregatorFactory[] aggregators,
-      @JsonProperty("granularitySpec") GranularitySpec granularitySpec
+      @JsonProperty("granularitySpec") GranularitySpec granularitySpec,
+      @JacksonInject ObjectMapper jsonMapper
```

Comment:
yes, in many places, for example see TaskToolboxFactory, WorkerCuratorCoordinator etc. And, I belived that was correct usage and it was useful to explicitly declare in situations where you wanted both (for example QueryResource)
Druid startup clearly binds same ObjectMapper instance to @Json and default, only @Smile has a different instance.
I believe it is OK.